app/src/main/java/org/simple/clinic/newentry/PatientEntryScreen.kt

```diff
@@ -108,21 +118,27 @@ class PatientEntryScreen(context: Context, attrs: AttributeSet) : RelativeLayout
   private val viewRenderer = PatientEntryViewRenderer(this)
 
   private val events: Observable<UiEvent> by unsafeLazy {
-    Observable.merge(
-        screenCreates(),
-        screenPauses(),
-        formChanges(),
-        saveClicks()
-    ).share()
+    Observable.merge(screenCreates(), formChanges(), saveClicks())
+        .compose(ReportAnalyticsEvents())
+        .share()
   }
 
   private val delegate by unsafeLazy {
+    val effectHandler = PatientEntryEffectHandler.createEffectHandler(
+        userSession,
+        facilityRepository,
+        patientRepository,
+        patientRegisteredCount,
+        this,
+        schedulersProvider
+    )
+
     MobiusDelegate(
         events.ofType(),
         PatientEntryModel.DEFAULT,
-        ::patientEntryInit,
-        ::patientEntryUpdate,
-        PatientEntryEffectHandler.createEffectHandler(userSession, facilityRepository, patientRepository, this, schedulersProvider),
+        ::patientEntryInit.toInit(),
```

Comment:
On second thought, I don't like the asymmetry between the init() and update() function invocations here. We should just make `::patientEntryInit` a concrete class and directly instantiate it here. What do you think?

Reply:
Or make the `PatientEntryUpdate` a function and convert it into a class using an extension like above.

Reply:
1. We could make `PatientEntryInit` a concrete class.
2. We can't make `PatientEntryUpdate` a class, because it needs access to `dobValidator` and `phoneNumberValidator`.

I'll go with 1 instead.